src/lib/registration/connect_helpers.rb

```diff
@@ -142,7 +148,12 @@ def self.catch_registration_errors(message_prefix: "",
         log.error "JSON parse error"
         # update the message when an old SMT server is found
         check_smt_api(e.message)
-        report_error(message_prefix + _("Cannot parse the data from server."), e.message)
+        msg = message_prefix + _("Cannot parse the data from server.")
```

Comment:
I'm not sure if we should retry on `JSON::ParserError`, the server might register the system properly (i.e. eating one of the customer subscription, but only the response is malformed or corrupted when transmitting over network).

But as this was mainly an issue with the SCCProxy then let the user decide, this should not happen with normal SCC...

Reply:
> I'm not sure if we should retry on `JSON::ParserError`, the server might register the system properly (i.e. eating one of the customer subscription, but only the response is malformed or corrupted when transmitting over network).

MMM in that case, when retrying the registration, it will be skipped as detected as already register, isn't it?

> 
> But as this was mainly an issue with the SCCProxy then let the user decide, this should not happen with normal SCC...

Yes, this looks more like the old registration server error than a normal SCC one, although the notes in Trello suggested to add the [Retry] option

Reply:
I'm fine with adding retry here, my point is that the user should know the consequences... But that's actually the case also for the timeout exception. It might happen that the server registers the client correctly but because of some network issue the client gets timeout...

Reply:
Exactly, and the timeout could happen at any time, that was what happened with the SCC qa proxy but there also the error is handled incorrectly.

Reply:
I removed the retry, but could add it back if wanted.